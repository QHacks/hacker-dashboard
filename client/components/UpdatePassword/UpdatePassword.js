import UpdatePasswordForm from '../utils/UpdatePasswordForm';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { actionCreators, selectors } from '../../HackerStore';
import { Link } from 'react-router-dom';

const { getIsPasswordUpdated } = selectors;
const { updatePasswordForReset } = actionCreators;

class UpdatePassword extends Component {

	handleUpdatePasswordForReset(values) {
		const { password } = values;
		const { hash } = this.props.match.params;

		this.props.updatePasswordForReset(hash, password);
	}

	render() {
		const { isPasswordUpdated } = this.props;

		if (isPasswordUpdated) {
			return (
				<div>
					<p>Your password has been updated!</p>
					<Link to="/login">Login now!</Link>
				</div>
			);
		}

		return (
			<div>
				<UpdatePasswordForm onSubmit={ this.handleUpdatePasswordForReset.bind(this) } />
			</div>
		);
	}
}

function mapStateToProps(state, ownProps) {
	return {
		...ownProps,
		isPasswordUpdated: selectors.getIsPasswordUpdated(state)
	};
}

export default connect(mapStateToProps, { updatePasswordForReset })(UpdatePassword);
