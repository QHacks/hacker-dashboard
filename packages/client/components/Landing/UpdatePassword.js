import React, { Component } from "react";
import { Link } from "react-router-dom";
import axios from "axios";
import ValidationError from "../ValidationError/ValidationError";
import ActionButton from "../ActionButton/ActionButton";
import { SERVER_HOST } from "../../Client";
import { blue } from "../../assets/colors";
import Alert from "../Alert/Alert";
import Landing from "./Landing";

class UpdatePassword extends Component {
  constructor(props) {
    super(props);

    this.state = {
      password: null,
      confirmPassword: null,
      alertShown: false,
      alert: {
        type: "",
        message: "",
        status: ""
      }
    };
  }

  showAlert(message, type, status = null) {
    this.setState({
      alert: {
        message,
        type,
        status
      },
      alertShown: true
    });

    setTimeout(() => this.setState({ alertShown: false }), 6000);
  }

  async updatePasswordForReset() {
    const { password } = this.state;
    const { resetHash } = this.props.match.params;

    try {
      await axios.post(`${SERVER_HOST}/oauth/updatePasswordForReset`, {
        password,
        resetHash
      });

      this.showAlert("Password has been successfully reset!", "success");
    } catch (err) {
      this.showAlert(err.response.data.message, "danger");
    }
  }

  render() {
    return (
      <Landing>
        <img
          src="../../assets/img/qhacks-wordmark-colored.svg"
          css="
            max-height: 40px;
          "
          alt="QHacks"
        />
        <h3
          css={`
            margin-top: 24px;
            color: ${blue};
            font-weight: 700;
          `}
        >
          Reset Password
        </h3>
        <p
          className="blurb"
          css="
            line-height: 1.6;
            margin-top: 16px;
            color: #8a929f;
          "
        >
          Please enter your new password.
        </p>
        <div
          css="
            margin-top: 64px;
          "
        >
          <input
            id="password"
            type="password"
            key="password"
            value={this.state.password}
            onChange={(e) => this.setState({ password: e.target.value })}
            placeholder="Enter your password"
          />
          <br />
          <ValidationError
            message={
              this.state.password !== null && this.state.password.length < 8
                ? "Password must be at least 8 characters"
                : ""
            }
          />
          <br />
          <input
            id="confirmPassword"
            type="password"
            key="confirmPassword"
            value={this.state.confirmPassword}
            onChange={(e) => this.setState({ confirmPassword: e.target.value })}
            placeholder="Confirm your password"
          />
          <br />
          <ValidationError
            message={
              this.state.confirmPassword !== null &&
              this.state.confirmPassword !== this.state.password
                ? "Passwords don't match!"
                : ""
            }
          />
        </div>
        <div
          css="
            margin: 30px 0;
          "
        >
          <Link className="landingLink" to="/login">
            Know you password? Login here!
          </Link>
        </div>
        <div>
          <ActionButton
            color="blue"
            onClick={() => this.updatePasswordForReset()}
          >
            Reset password
          </ActionButton>
        </div>
        <div
          css="
          margin: 30px 0px 0px;
        "
        >
          {this.state.alertShown ? <Alert {...this.state.alert} /> : ""}
        </div>
      </Landing>
    );
  }
}

export default UpdatePassword;
