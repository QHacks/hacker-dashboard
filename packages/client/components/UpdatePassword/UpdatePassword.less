@import (less) '../../assets/semantic-ui/site/globals/site.variables';

.update-container {
	display: flex;
	flex-direction: column;
	align-items: center;
	min-height: 100vh;
}

.update-header {
	align-items: center;
	display: flex;
	flex-direction: column;
	margin-top: 40px;
	margin-bottom: 30px;
}

.update-footer {
	margin-top: 20px;
	margin-bottom: 40px;
}

.update-form-container {
	display: flex;
	flex-direction: column;
	padding: 30px 20px;
	max-width: 600px;
	width: 100%;
}
