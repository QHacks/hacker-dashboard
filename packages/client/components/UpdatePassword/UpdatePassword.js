import { Divider, Header, Message } from "semantic-ui-react";
import { actionCreators, selectors } from "../../HackerStore";
import { UpdatePasswordForm } from "../Forms";
import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import "./UpdatePassword.less";

const {
  getIsPasswordUpdated,
  getIsPasswordUpdatedError,
  getIsPasswordUpdatedLoading
} = selectors;
const { updatePasswordForReset } = actionCreators;

class UpdatePassword extends Component {
  handleUpdatePasswordForReset(values) {
    const { password } = values;
    const { hash } = this.props.match.params;

    this.props.updatePasswordForReset(hash, password);
  }

  renderPasswordUpdateHeader() {
    const { isPasswordUpdated } = this.props;

    return (
      <div className="update-header">
        <img
          src={require("../../assets/img/qhacks-tricolor-logo.svg")}
          className="qhacks-logo"
        />
        <Header
          as="h2"
          content="Update Password"
          color="red"
          textAlign="center"
          className="form apply header"
        />
        {isPasswordUpdated ?
          null :
          (
            <p>
              Please enter your new password to complete the reset password
              process.
            </p>
          )}
      </div>
    );
  }

  renderPasswordUpdateForm() {
    const {
      isPasswordUpdated,
      isPasswordUpdatedError,
      isPasswordUpdatedLoading
    } = this.props;
    if (isPasswordUpdated) {
      return (
        <Message
          success
          size="small"
          header="Password Update Successful!"
          content="Congratulations your password has been updated. Please proceed to login."
        />
      );
    }
    return (
      <UpdatePasswordForm
        onSubmit={this.handleUpdatePasswordForReset.bind(this)}
        updateError={isPasswordUpdatedError}
        updateLoading={isPasswordUpdatedLoading}
      />
    );
  }

  renderPasswordUpdateFooter() {
    return (
      <div className="update-footer">
        <Divider />
        <p className="fontSize-medium textAlign-center">
          Wrong place? <Link to="/login">Login here</Link>
        </p>
      </div>
    );
  }

  render() {
    return (
      <div className="update-container">
        <div className="update-form-container">
          {this.renderPasswordUpdateHeader()}
          {this.renderPasswordUpdateForm()}
          {this.renderPasswordUpdateFooter()}
        </div>
      </div>
    );
  }
}

function mapStateToProps(state, ownProps) {
  return {
    ...ownProps,
    isPasswordUpdated: getIsPasswordUpdated(state),
    isPasswordUpdatedError: getIsPasswordUpdatedError(state),
    isPasswordUpdatedLoading: getIsPasswordUpdatedLoading(state)
  };
}

export default connect(mapStateToProps, { updatePasswordForReset })(
  UpdatePassword
);
