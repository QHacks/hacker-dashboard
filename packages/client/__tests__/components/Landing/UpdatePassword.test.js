import React from "react";
import UpdatePassword from "../../../components/Landing/UpdatePassword";
import { shallow } from "enzyme";

describe("UpdatePassword", () => {
  it("renders correctly", () => {
    expect(shallow(<UpdatePassword />)).toMatchSnapshot();
  });
});
