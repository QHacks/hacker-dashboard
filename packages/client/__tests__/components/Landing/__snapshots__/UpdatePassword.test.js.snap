// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatePassword renders correctly 1`] = `
<Landing>
  <img
    alt="QHacks"
    className="css-zl7mak"
    src="../../assets/img/qhacks-wordmark-colored.svg"
  />
  <h3
    className="css-xqvf5y"
  >
    Reset Password
  </h3>
  <p
    className="css-o4thmd blurb"
  >
    Please enter your new password.
  </p>
  <div
    className="css-h8m1q6"
  >
    <input
      id="password"
      key="password"
      onChange={[Function]}
      placeholder="Enter your password"
      type="password"
      value={null}
    />
    <br />
    <ValidationError
      message=""
    />
    <br />
    <input
      id="confirmPassword"
      key="confirmPassword"
      onChange={[Function]}
      placeholder="Confirm your password"
      type="password"
      value={null}
    />
    <br />
    <ValidationError
      message=""
    />
  </div>
  <div
    className="css-1tyn5zp"
  >
    <Link
      className="landingLink"
      replace={false}
      to="/login"
    >
      Know you password? Login here!
    </Link>
  </div>
  <div>
    <ActionButton
      color="blue"
      onClick={[Function]}
    >
      Reset password
    </ActionButton>
  </div>
  <div
    className="css-17170mj"
  />
</Landing>
`;
